src/module_3/Module3_Milestone1.md

```diff
@@ -0,0 +1,1825 @@
+daily_orders = filtered_data.groupby('order_date').order_id.nunique()
+daily_orders.head()
+```
+
+
+
+
+    order_date
+    2020-10-05     3
+    2020-10-06     7
+    2020-10-07     6
+    2020-10-08    12
+    2020-10-09     4
+    Name: order_id, dtype: int64
+
+
+
+
+```python
+plt.plot(daily_orders, label="Daily orders")
+plt.title("Daily Orders")
+```
+
+
+
+
+    Text(0.5, 1.0, 'Daily Orders')
+
+
+
+
+    
+![png](Module3_Milestone1_files/Module3_Milestone1_21_1.png)
+    
+
+
+**Important**: It's evident that there is a significant time-based aspect, signaling changes in patterns and business evolution over time. Consequently, it is prudent to implement a temporal split. This approach not only captures the dynamic nature of the data but also ensures that the same order does not appear in both the training and testing sets, thereby preventing information leakage.
+
+
+```python
+cum_sum_daily_orders = daily_orders.cumsum() / daily_orders.sum()
+
+train_val_cut = cum_sum_daily_orders[cum_sum_daily_orders <= 0.7].idxmax()
+val_test_cut = cum_sum_daily_orders[cum_sum_daily_orders <= 0.9].idxmax()
+
+print("Train set from: ", cum_sum_daily_orders.index.min())
+print("Train set until: ", train_val_cut)
+print("Validation set from: ", val_test_cut)
+print("Test set until: ", cum_sum_daily_orders.index.max())
+```
+
+    Train set from:  2020-10-05 00:00:00
+    Train set until:  2021-02-04 00:00:00
+    Validation set from:  2021-02-22 00:00:00
+    Test set until:  2021-03-03 00:00:00
+
+
+It is important to consider what is the minimum business cicle, that is, the minimum temporal window that allows us to capture the complete dynamics of the data we want to model. Even though the validation and test splits don't comprise many days, we consider these should be sufficient for our case.
+
+
+```python
+train_data = filtered_data[filtered_data.order_date <= train_val_cut]
+val_data = filtered_data[(filtered_data.order_date > train_val_cut) & (filtered_data.order_date <= val_test_cut) ]
+test_data = filtered_data[filtered_data.order_date > val_test_cut]
+```
+
+Divide into features and target:
+
+
+```python
+def feature_target_split(df, target):
+    X = df.drop(columns=[target])  # Drop the target column to create the features DataFrame
+    Y = df[target]   # Target variable we want to predict
+    return X, Y
+
+X_train, Y_train = feature_target_split(train_data, target_col)
+X_val, Y_val = feature_target_split(val_data, target_col)
+X_test, Y_test = feature_target_split(test_data, target_col)
+```
+
+### 2. Defining a Baseline model
+
+In order to be able to decide if a ML model generates any value, we need to compare it against baselines that do not require training. In this case, we will use the global popularity feature as baseline, that is, the product will be predicted to be bought the more popular it is.
+
+
+```python
+def plot_curves(y_true, y_pred, curve_type="both", dataset_type="train"):
```

Comment:
In notebooks is ok not to be super picky with code quality. But, given that you are paying a lot of attention on commenting the code, I think here it will be of much more value to have a docstring explaining  which curves are being plot and what is curve_type and dataset_type and which values are allowed than any other comment